arroyo-connectors/src/webhook.rs

```diff
@@ -0,0 +1,207 @@
+use std::{convert::Infallible, time::Duration};
+
+use anyhow::{anyhow, bail};
+use arroyo_rpc::OperatorConfig;
+use arroyo_types::string_to_map;
+use axum::response::sse::Event;
+use reqwest::{header::HeaderMap, header::HeaderName, header::HeaderValue, Client, Request};
+use serde_json::json;
+use tokio::sync::mpsc::Sender;
+use typify::import_types;
+
+use arroyo_rpc::types::{ConnectionSchema, ConnectionType, TestSourceMessage};
+use serde::{Deserialize, Serialize};
+
+use crate::{pull_opt, Connection, EmptyConfig};
+
+use super::Connector;
+
+const TABLE_SCHEMA: &str = include_str!("../../connector-schemas/webhook/table.json");
+
+import_types!(schema = "../connector-schemas/webhook/table.json");
+const ICON: &str = include_str!("../resources/webhook.svg");
+
+pub struct WebhookConnector {}
+
+impl WebhookConnector {
+    fn construct_test_request(config: &WebhookTable) -> anyhow::Result<(Client, Request)> {
+        if let Err(e) = reqwest::Url::parse(&config.endpoint) {
+            bail!("invalid endpoint '{}': {:?}", config.endpoint, e)
+        };
+
+        let headers: anyhow::Result<HeaderMap> =
+            string_to_map(config.headers.as_ref().map(|t| t.0.as_str()).unwrap_or(""))
+                .expect("Invalid header map")
+                .into_iter()
+                .map(|(k, v)| {
+                    Ok((
+                        TryInto::<HeaderName>::try_into(&k)
+                            .map_err(|_| anyhow!("invalid header name {}", k))?,
+                        TryInto::<HeaderValue>::try_into(&v)
+                            .map_err(|_| anyhow!("invalid header value {}", v))?,
+                    ))
+                })
+                .collect();
+
+        let client = reqwest::ClientBuilder::new()
+            .default_headers(headers?)
+            .timeout(Duration::from_secs(5))
+            .build()
+            .map_err(|e| anyhow!("could not construct HTTP client: {:?}", e))?;
+
+        let req = client
+            .post(&config.endpoint)
+            // TODO: use the schema to construct a correctly-formatted message
```

Comment:
Do we need to do this before merging?

Reply:
I think it's a nice-to-have (if the receiving endpoint is expecting a particular format, it may return an error during testing) but the user can still decide to create the source.